File: swift/basics/fizzbuzz.swift
// Print FizzBuzz for 1 through 100, choosing the label with a switch over a tuple.
// Run: swift fizzbuzz.swift

func fizzBuzzLabel(_ n: Int) -> String {
    switch (n % 3, n % 5) {
    case (0, 0): return "FizzBuzz"
    case (0, _): return "Fizz"
    case (_, 0): return "Buzz"
    default: return String(n)
    }
}

for n in 1...100 {
    print(fizzBuzzLabel(n))
}

File: haskell/basics/fizzbuzz.hs
-- Print FizzBuzz for 1 through 100, choosing the label with guards over the remainders.
-- Run: runghc fizzbuzz.hs

label :: Int -> String
label n
  | n `mod` 15 == 0 = "FizzBuzz"
  | n `mod` 3 == 0 = "Fizz"
  | n `mod` 5 == 0 = "Buzz"
  | otherwise = show n

main :: IO ()
main = mapM_ (putStrLn . label) [1 .. 100]
